### ISSUE TITLE
Unable to Use Elasticache

### ISSUE THREAD

**Issue by chrisnellis** [awspring/spring-cloud-aws#221]
**Type**: Bug

**Component**:
ElastiCache-Memcached

**Describe the bug**
I'm running Spring Boot 2.6.1 in Fargate. The ECS execution role already has the `elasticache:DescribeCacheClusters` permission (as required by [the docs here](https://docs.awspring.io/spring-cloud-aws/docs/2.3.2/reference/html/index.html#iam-permissions-5)). I'm already successfully pulling configs from the parameter store, but when I'm trying to configure Spring to use an Memcached cluster in Elasticache for caching, Spring doesn't recognize it as a cache source. When I ran the code with Trace level debugging, I saw the following message as it successfully started:
```Condition OnClassCondition on io.awspring.cloud.autoconfigure.cache.ElastiCacheAutoConfiguration did not match due to 
@ConditionalOnClass did not find required class 'com.amazonaws.services.elasticache.AmazonElastiCache'

         - @ConditionalOnClass did not find required class 'com.amazonaws.services.elasticache.AmazonElastiCache' (OnClassCondition)
```

After extensive searching, I then tried to add in `com.amazonaws.aws-java-sdk-elasticache.1.12.129` as a dependency, which didn't help either. It yielded the following error:
java.lang.RuntimeException: Error creating cache

```Caused by: java.lang.IllegalArgumentException: No CacheFactory configured for engine: memcached ```

**Sample**
Relevant dependencies:
```xml
        <!-- (Needed to pull configs from parameter store) -->
        <dependency>
            <groupId>io.awspring.cloud</groupId>
            <artifactId>spring-cloud-starter-aws-parameter-store-config</artifactId>
            <version>2.3.3</version>
        </dependency>

        <dependency>
            <groupId>io.awspring.cloud</groupId>
            <artifactId>spring-cloud-starter-aws</artifactId>
            <version>2.3.3</version>
        </dependency>

        <dependency>
            <groupId>io.awspring.cloud</groupId>
            <artifactId>spring-cloud-aws-context</artifactId>
            <version>2.3.3</version>
        </dependency>

        <dependency>
            <groupId>com.amazonaws</groupId>
            <artifactId>aws-java-sdk-elasticache</artifactId>
            <version>1.12.129</version>
        </dependency>
```

Application-cloud.yml:
```yml
spring:
  config:
    import: "aws-parameterstore:"

cloud:
  aws:
    credentials:
      data:
        enabled: true
    elasticache:
      clusters:
        - name: memcached-cluster-name
          expiration: 10
      enabled: true
```

Code invoking the cache:
```java
@Service
public class Service {

    private CacheManager cacheManager;

    private ReallySlowWebService reallySlowWebService;

    private Cache cache;

    @PostConstruct
    private void init() {
        cache = cacheManager.getCache("memcached-cache-name");
    }

    public Service(ReallySlowWebService reallySlowWebService, CacheManager cacheManager) {
        this.cacheManager = cacheManager;
        this.reallySlowWebService = reallySlowWebService;
    }

    //main entry point
    public Mono<FormattedServiceResponse> getFormattedResponse(RequestMetadata requestMetadata) {
        FormattedServiceResponse cachedResponse = cache.get(requestMetadata.getInternalUserID(), FormattedServiceResponse.class);
        if (cachedResponse != null) {
            return Mono.just(cachedResponse);
        } else {
            return this.getFormattedResponseFromService(requestMetadata);
        }
    }

    private Mono<FormattedServiceResponse> getFormattedResponseFromService(RequestMetadata requestMetadata) {
        Mono<ReallySlowServiceResponse> serviceResponseMono = reallySlowWebService.callReallySlowWebService(requestMetadata);
        return serviceResponseMono.flatMap(reallySlowServiceResponse -> Mono.just(new FormattedServiceResponse(reallySlowServiceResponse)))
                .doOnNext(formattedServiceResponse -> this.cache.put(requestMetadata.getInternalUserID(), formattedServiceResponse));
    }
}
```

(I recognize that calling cacheManager in this way may be blocking, but it was the only way I could cache the actual response object without invoking extra calls to the really slow web service)

**Comment by maciejwalkowiak** [comment]
Adding dependency to:

```xml
<dependency>
	<groupId>com.amazonaws</groupId>
	<artifactId>elasticache-java-cluster-client</artifactId>
</dependency>
```

should solve this issue. This should be included in docs. 

Btw, we are considering removing support for Memcached in Spring Cloud AWS 3.0. Why have you chosen Memcache over Redis?

**Comment by chrisnellis** [comment]
Hello.

I'm sorry I missed that. To make a deadline, I ended up just using the elasticache client described above and ditching this framework for the use case. (So I'll close this issue upon adding this comment since I'm no longer pursuing this path)

After reviewing [the feature matrix from AWS](https://aws.amazon.com/elasticache/redis-vs-memcached/), we realized Memcached was sufficient for our use case. 

If we end up migrating to Redis, I'll reconsider using this framework.

Thanks for your reply, @maciejwalkowiak !
